pydata_center/monitoring/graylog.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import os
+
+import graypy
+from celery import shared_task
+
+graylog_logger = logging.getLogger('graylog_logger')
+graylog_logger.setLevel(logging.DEBUG)
+
+graylog_host = os.getenv('GRAYLOG_HOST', '127.0.0.1')
+graylog_gelf_tcp_port = int(os.getenv('GRAYLOG_GELF_TCP_PORT', '12201'))
+
+tcp_handler = graypy.GELFTCPHandler(graylog_host, graylog_gelf_tcp_port)
```

Comment:
Avoid Handler Duplication on Reload (Especially in Django)
Every time the file is imported (e.g., via Django auto-reload), the handler gets re-added. This leads to duplicate log entries.

Fix: Check if handler already exists:

if not any(isinstance(h, graypy.GELFTCPHandler) for h in graylog_logger.handlers):
    tcp_handler = graypy.GELFTCPHandler(graylog_host, graylog_gelf_tcp_port)
    graylog_logger.addHandler(tcp_handler)